### ISSUE TITLE
EPIC: Unpaid Fee Flag on Permit Issuance

### ISSUE THREAD

**Issue by atdservicebot** [cityofaustin/atd-data-tech#1518]
### We are adding a fee validation & warning message for unpaid ROW fees. 
- See epic issues for work performed, this issue pertains only to scoping.

**Comment by amenity** [comment]
Sree and Raj will send list of questions to @SurbhiBakshi and she will forward to Paloma and Ryan.

**Comment by SurbhiBakshi** [comment]
ROW-LIT Meeting scheduled to gather more details on 04/21

**Comment by TracyLinder** [comment]
We need an update from LIT on this one.

**Comment by Nadin-Nader** [comment]
@TracyLinder Sree is waiting for UAT on this but on DEV. I dont know if Kim has access to dev but i will schedule a meeting with her and she can use my DEV if need be

**Comment by Nadin-Nader** [comment]
Scheduled meeting for 5/19/2020 @10am with Kim

**Comment by Nadin-Nader** [comment]
@TracyLinder This UAT was done along with the UAT for #1951  and is located in the second half of the video in the link: [https://web.microsoftstream.com/video/c540fe12-2fa6-4f21-b56f-6aad97c00079](url) This UAT starts at 22 minutes into the video.

The main issue is that the people records are not checked to see if there were any unpaid fees on any of the people records, it also doesnt throw out a flag or warn the user. It allows them to continue to issue the permit.
Kim Used the EX folder and Austin Underground as the Right of Way Contractor (ROWID: 5014638). They had $2,743.50 in fees but they were from 2016.
**Side note- They wanted the system to check the people record from October 01, 2013 til present or all dates for any unpaid fees**
Kim created an EX folder with a sub of "Private Contractor", was able to type in the Description and Conditions field For the facility/Owner info, PVT Contractor was selected. After selecting everything that was needed and Kim accepted the permit.....the system created the folder without at least a warning saying that there were unpaid fees in Austin Underground's People records.
@TracyLinder  do you want to let LIT know that this is not working? or do you want me to?

**Comment by Nadin-Nader** [comment]
Sree is waiting on feedback from an email she sent out on April 30th, 2020 (I did not receive it)
![image](https://user-images.githubusercontent.com/63309637/82484605-c4c5db00-9a9f-11ea-8d32-99f06be418b7.png)

I then sent an email to Sree asking her to send me a copy of the email to see if i can help.
Waiting on response from Sree or from the Stakeholders.

**Comment by TracyLinder** [comment]
Sree has the feedback she needs on this. She has everything she needs to continue working.

**Comment by TracyLinder** [comment]
**From Sree:**
Hi Tracy,

Right now we should query any fees due is not zero right for the associated people folders; Do I also need to consider Unbilled fees due???

**My Response:**
Hi Sree,

You do not need to consider unbilled fees. Only considered billed fees on the specified folder types.

**Comment by TracyLinder** [comment]
**Email w/ Sree Today:**
Hi Sree,

• For ECV Folder there is no ISSUE procedure at all though there is  a button in Folder. Shall we create a ISSUE procedure to have the Undue Payment flag check? And Folder status update ? Please confirm. – jjj 
        • These questions are a bit too technical for me unfortunately. What I can confidently say is that we want this fee check to look for unpaid bills on ECV folder. I don’t know the best technical solution for this. Nate – do you have any insight? 
• An Applicant or Right of a Contractor of EX folder, associated with another LC/GF folder as a Billed to or Owner (other peopletype), then do we need to consider the unpaid fee of LC/GF Folder or NOT?
        •  In this scenario, yes, ideally the stakeholders want that fee to be considered. 

Thank You,

**Comment by TracyLinder** [comment]
See #2893 for more info about next steps. Stakeholder input is needed to help design the best solution. Pending stakeholder meeting, 6/4 @ 1:30.

**Comment by Nadin-Nader** [comment]
@TracyLinder Notes from #2893:

Meeting Video Link: https://web.microsoftstream.com/video/8632862c-e33a-43ea-831a-0ada6a570d62

Sree started by trying to issue a permit and explained that the first check violation is the current warning that comes up and that this warning is a page that they cannot make changes to because it will involve CTM involvement. This only checks in the current folder.

Sree continued through the first warning and then the new unpaid fees warning that shows the amounts and people record came up. When the warning is accepted, the process continues, issues the permit and makes it Active. From here the reviewer can do the research to find out what fees are needed.

Paloma explained that this does not meet their acceptance criteria as they need to be able to have the ability to issue or NOT issue the permit if there are unpaid fees in any folder that the people record is associated with.

I then proposed the options that are allowed by AMANDA and that we (Tracy, Sree and myself) came up with:

Set monetary limit for hard stop (permit not issued) and any fees less have warning (Permit is issued)
Sree mentioned that this would work with the Issue button

Paloma wanted to know if this option gives them the ability to "Pause" the issuance if the monetary value was set low, while they investigated; then they could come back and issue the permit. I explained that AMANDA didnt work that way and that the process of issuing would have to restart.

Paloma said that this option would not work either

-Hard stop (Permit not issued)
Didnt work for them

Warning only (Permit is issued)
Didnt work for them

Change process to a folder Process attempt (This is a good one but will it hinder workflow or add value?)

Paloma mentioned that they currently go to the people tab and research each record

Sree suggested to manually insert a new attempt "Approve it Anyway" in the process tab on the last process the reviewer uses or has in the folder. It will still issue and approve but give the reviewer the option to issue or not. "Approved" attempt will throw the hard stop error and not issue the permit.

This Option will make the user give 2 approvals instead of one in the process.

This will still give the error with the amounts owed

LAST PROCESSES USED: EX = Intake Review Process, RW = Intake Review
Wont work for DS since there are no processes on the DS folder unless we add a Process

They decided to go with the last option knowing that the DS folder will need a new process created in order for this to work.
**

On the DS folder maybe use current process and/or add the Right of Way Admin process but only to reactivate permits but not on the first issuance of the permit. New DS folder will have a Web Acceptance process to it as well as the one added.

DS goes straight to issue if it doesnt come from the Web. When it does it gets the Web Acceptance process and then they issue. (They select the Approved attempt then issue button)

Issue with this is that not all DS are applied for online so they would have to add a new Process for Existing DS ones, they just select the Issue button from the folder tab (no process on this)

Big problem with adding the process to the DS Web Acceptance; is that you have to have an Internet User and it cannot be changed or added.
50850 DS INTAKE AND REVIEW could be a possible solution for this. Only complication is that they dont know for certain how it would affect the folder. Sree said this would require more testing to see what it does.

The change can be applied to the EX, LC, RW folders and then later the DS once it is researched more.

**> The final decision of this issue was to Ice Box it (put on hold) due to the renewals and extensions not having the same processes as the first issuance processes so this will need to have the changes made to the first issuance of folder and then the renewals.

This is a lot of effort on the Reviewer side to research and a lot on our end to use the option considered. Paloma will need to speak to Ryan Mooney and her team to see if this can wait til after AMANDA V7 to use the first warning where they get the option to stop or continue anyway instead or creating all these processes or attempts. Paloma will invite myself and Tracy to the meeting.**

**Comment by Nadin-Nader** [comment]
@TracyLinder 
Email from Paloma:

> From: Amayo-Ryan, Paloma <Paloma.Amayo-Ryan@austintexas.gov> 
> Sent: Thursday, June 4, 2020 3:39 PM
> To: Nader, Nadin <Nadin.Nader@austintexas.gov>; Linder, Tracy <Tracy.Linder@austintexas.gov>; Sreelatha.b@launchitcorp.com
> Cc: Mooney, Ryan <Ryan.Mooney@austintexas.gov>; Perez, Kimberly <Kimberly.Perez@austintexas.gov>
> Subject: INFO: Unpaid Fees Flag on Permit Issuance #1518
> 
> Good afternoon all,
> 
> I had an opportunity to confer with Ryan Mooney and we have come to the conclusion that we should pause further development of this issue until AMANDA v7 has been implemented.  We’re hopeful that this will lead to a better solution.
> 
> Thank you for your efforts thus far!
> Paloma
>

**Comment by TracyLinder** [comment]
### DTS Conversation for Scoping Prioritization 3/26/24

- The current AC specifies features that don't exist, and need to be revisited.
- The business need is still clear; ROW analysts want to know if the applicant has unpaid TPW fees before they issue a permit. 
- Adding the validation to the folder issuance procedure is feasible, with a simple warning
- Vish will pick this up and look at it with fresh eyes.

**Comment by vschandramouli** [comment]
Adding @rajeshlaunch to review the AC and provide potential solutions. 

CC @TracyLinder

**Comment by rajeshlaunch** [comment]
Amanda does have an option to show warning messages which won't prevent the users to proceed with next steps. However, it doesn't provide options of taking actions against the warning message through a button click like it was mentioned in the above requirement. There are few ways we can try to implement the requirement,

Solution 1:

1. Query the database with specifics filter to identify whether any unpaid fees exists for the given criteria.
2. Show the default warning message to user at the time of issuance.
3. Add a process "Unpaid fees" during issuance if there any unpaid fees exists with 2 attempts, "Generate and Send pending fees invoice", "Completed".
4. "Unpaid Fees" process will be assigned to a user and the user can open the process and the attempt "Generate & Send pending fees invoice", this will generate the invoice shown in the requirement and send it to the customer.
5. Once the fee is collected, they can use "Completed" attempt to close the process.
6. 

Solution 2:
1. Instead of warning message, we can show error message which will prevent the issuance of the permit if there are any unpaid fees.
2. All other steps in Solution 1 can be performed to collect the fees and then allow user to issue the permit.

@vschandramouli @TracyLinder We may need to further triage the solution to proceed with next steps.

**Comment by TracyLinder** [comment]
### Refinement with ROW during Friday's meeting 4/12

- No to a hard stop, yes to a warning message
- There are scenarios where ROW issues the permit when there are unpaid ROW fees
- New folder process idea: Not liking the idea of a new folder process; would rather add a new attempt to an existing process.
- When during the permit review would ROW like to run the unpaid fee check? - 


**Preferred Solution:**
1. System performs unpaid fee validation during folder issuance, with soft warning 
2. Add the unpaid fee validation to existing process attempts that are already being used by staff. TBD which attempts on which folders this feature should be added to. 

**Next Steps:**
- [ ] Determine which existing process attempts to add the validation to.

**Comment by TracyLinder** [comment]
**Process attempts to add the fee validation to (from Kim)**

<span><span class="ui-provider ed bdw azt bsd bse bsf bsg bsh bsi bsj bsk bsl bsm bsn bso bsp bsq bsr bss bst bsu bsv bsw bsx bsy bsz bta btb btc btd bte btf btg bth bti" dir="ltr">
Folder Type | SubType | Process | Attempt
-- | -- | -- | --
RW | Street Vendor | Vendor Permitting | Update Issue and Expiry Dates
RW | Valet Operator | Valet Operator Permitting | Update Issue and Expiry Dates
RW | Valet Parking Annual | Annual Valet Permitting | Update Issue and Expiry Dates
RW | Temporary Right of Way Use Permit | Permit Extension | Manual Extension or Extension Granted
DS | Any | Right Of Way Admin | Reactivated
EX | Any | Right Of Way Admin | Manual Extension or Extension Granted
PSM | Any | Right Of Way Admin | Manual Extension

</span></span>

**Comment by TracyLinder** [comment]
### DTS-AMANDA Refinement Convo

- We will move forward with configuring the fee validation check on the above process attempts
- Discuss with ROW creating an unpaid fee report that could be configured on the folder or the report library. This would replace the check that PLA's are doing and provide the applicant with documentation.

**Comment by TracyLinder** [comment]
### Friday review convo with ROW about potential report:
1. It could possibly save some time. Not a bad idea.

**Comment by rajeshlaunch** [comment]
AC:

1. Please pull the unpaid fees during below process and attempts were made. 
<span><span class="ui-provider ed bdw azt bsd bse bsf bsg bsh bsi bsj bsk bsl bsm bsn bso bsp bsq bsr bss bst bsu bsv bsw bsx bsy bsz bta btb btc btd bte btf btg bth bti" dir="ltr">
Folder Type | SubType | Process | Attempt
-- | -- | -- | --
RW | Street Vendor | Vendor Permitting | Update Issue and Expiry Dates
RW | Valet Operator | Valet Operator Permitting | Update Issue and Expiry Dates
RW | Valet Parking Annual | Annual Valet Permitting | Update Issue and Expiry Dates
RW | Temporary Right of Way Use Permit | Permit Extension | Manual Extension or Extension Granted
DS | Any | Right Of Way Admin | Reactivated
EX | Any | Right Of Way Admin | Manual Extension or Extension Granted
PSM | Any | Right Of Way Admin | Manual Extension

</span></span>
2.  Logic to pull the unpaid fee is as follows: Query all unpaid fees for each people record on the people tab. Of those fees, exclude any not in folder types RW, EX, LC, ECV, GF, or DS. Of those fees, exclude any created prior to 10/1/2013. Now you have a list of qualifying fees.
3. If there are any outstanding fees returned from the above query, show the warning message as "A People record from this folder has UNPAID fees!"
4. Allow the user to proceed further after they click "OK" on the warning message.

@vschandramouli Please review the AC and move the card to next stage.

**Comment by vschandramouli** [comment]
Reviewed and updated AC, moving to Backlog.

**Comment by TracyLinder** [comment]
### Removed from description/AC
_Pasting here for reference._ 

## This issue was on hold until AMANDA V7 Implementation 

## Unpaid Fee Flag
_AKA # 46 Unpaid Fee Flag on Permit Issuance (Audit Request)_

**Stakeholders:** Paloma, Kim, Mooney

**User Story:** As a ROW permit analyst (PLA), I need to know if any of the folder people have unpaid ROW fees BEFORE the permit is issued, so I can place the permit on hold and attempt to collect the outstanding fee(s).

**Current State:**
1. The PLA, prior to permit approval, performs manual research to identify any unpaid ROW fees for ALL people listed on the people tab (Contractor, applicant, contact, whoever). 
    - They do not collect fees billed before 10/1/2013.
2. If no outstanding fees are identified, the permit is issued. End process. 
3. If outstanding ATD fees are identified, the PLA provides the outstanding fee information to the person/entity responsible for payment. 
    - So if there are 3 unique people/entities on the permit, each entity will be provided with their unique fee information. 
    - The people/entities are provided with instructions for paying, all outstanding bill numbers, and the total for each bill number. 
4. The PLA adds a note to the folder description: "Pending Outstanding Fee Payment **TL"
5. The customer will pay the fees, either online or in person. 
6. The customer notifies PLA of payment
    - When this step is missed, it leads to customers working in the ROW without an approved permit
7. PLA issues the permit. End Process. 

**Future State:**
1. Folder issuance is attempted, either by a user from the folder tab, or by a system procedure like post payment.
    - We need to be sure any new procedure doesn't cause payment update failures when the fees are paid online. (Put simply, lets learn from the issues caused by the Post Payment upgrade.)
    - This feature may have to be split into 2 features, one for running this procedure when issuance is initiated by the user from the folder tab, and another for running this procedure when issuance is initiated by a post payment procedure. **Will need LIT feedback on this.**
2. The system performs the unpaid fee verification
    - See # 1 in Current State; the system needs to do this instead of the PLA
    - Query all unpaid fees for each people record on the people tab. Of those fees, exclude any not in folder types RW, EX, LC, ECV, GF, or DS. Of those fees, exclude any created prior to 10/1/2013. **Now you have a list of qualifying fees.** 
3. If **qualifying fees** are found, an alert window opens informing the PLA that unpaid fees were found. 
    - The window will display this text: A people record related to this permit has unpaid ROW fees!
        - **Stakeholder input needed.** Please provide feedback on the proposed content of the alert window:    
![image](https://user-images.githubusercontent.com/49930868/74903386-60935e00-536e-11ea-9440-7aa50dc414c4.png)
    - The window will facilitate the following actions:
        - Allow the user to click a button to continue with permit issuance
        - Allow the user to click a button to abort the permit issuance, and view information related to the qualifying fees
        - The people invoice list view is an example of how we might provide the user with data about the qualifying fees:
![image](https://user-images.githubusercontent.com/49930868/74903558-e0212d00-536e-11ea-968f-a603023fc067.png)
4. PLA will provide the unpaid fee information (see list view above) to the customer, and note 'Pending Outstanding Fee Payment**' in the folder description.
    - If it's possible to email the 'list view' of unpaid fees to the customer from the window in AMANDA, that would be ideal **Will need LIT feedback on this.**
5. Customer pays
    - Jump to # 8 if customer does not pay
6. AMANDA notifies the PLA (or perhaps a resource task list?) that the permit needs to be issued **Stakeholder input needed. How would they most like to be notified?**
7. PLA issues the permit. End Process.
8. **Stakeholder input needed.** Will ROW need/want to know that these fees are still unpaid after a certain amount of time? For example, a monthly report emailed by the system with info about all 'fee flag alerts' that occurred in the previous month; only include info for fees that are still unpaid.
    - This would be another story/feature however LIT needs to be aware of the reporting requirement when developing the fee flag feature. 
------


Old Description: Requesting an alert to notify analysts when there are unpaid fees affiliated with any people records tied to a folder. Something similar to the current alert received when an analyst attempts to issue a permit that has violations affiliated with an address records tied to a folder.

Request Date: 2018-11-29 17:12:00

Request ID: DTS18-103080

Status: Backlog

Customer Priority: 0

Level of Effort: Small Project

DTS URL: https://atd.knack.com/dts#service-requests/view-issue-details/5c00727bd1bf2c4a9f4d268c



*Migrated from [atd-amanda #46](https://github.com/cityofaustin/atd-amanda/issues/46)*
